### PR TITLE
Community Destroyer - Harder, Better, Faster, Stronger!

### PR DESCRIPTION
**Pull request by soof-golan**
# TL;DR: What does this do?

Allows CEA to decide that all community posts only get 1/2 as much score, for purposes of sorting posts on the front page.

## Wait a minute, this looks familiar 🤨

Correct! This is a successor of the first version of [community-destroyer](https://github.com/centre-for-effective-altruism/EAForum/pull/249)

## It's fully generic!

Any tag document with a `scoreMultiplier` field will modify the frontpage post sorting score.

That mean CEA can decide to boost (or downgrade) frontpage posts with certain tags with ease.

## How does it work?

We lookup the tag documents per post, and multiply the post score by their `scoreMultiplier` field.

# Tests

## Automatic tests (jest)

Still none of that, sorry. It seemed hard to figure out how to test this (but if someone will give me a hand, I'll happily add a test)

## Manual tests

Real test against the dev DB coming soon.
I'm waiting for a green light from CEA to perform that test.

### No code changes

image

### With a default tag multiplier of `1` (no-op)

image

### Each tag grants you 1000X score

image

### Each tag grants you 1000X LESS score

image

### With a multiplier of `0` (remove score sorting altogether)

image

# Authors

This was pair programmed with [Yonatan Cale](https://github.com/hibukki)

# Before Merging

This needs to be tested against the dev DB with actual score modifiers.

# Open to pushback and changes of course!

This felt more like "figure out what's going on" than "tidy development", perhaps we missed something

### CODE REVIEW

File: packages/lesswrong/lib/collections/posts/views.ts [hunk 1]
```diff
@@ -278,30 +279,33 @@ function filterSettingsToParams(filterSettings: FilterSettings): any {
   }
   
   const tagsSoftFiltered = _.filter(filterSettings.tags, t => (t.filterMode!=="Hidden" && t.filterMode!=="Required" && t.filterMode!=="Default" && t.filterMode!==0));
-  let scoreExpr: any = null;
-  if (tagsSoftFiltered.length > 0 || frontpageSoftFilter.length > 0) {
```
Comment (hibukki) [inline packages/lesswrong/lib/collections/posts/views.ts:282]: Reviewers, notice:
Removing this `if` is supposed to fix an existing bug in the codebase:

The bug is:
If `tagsSoftFiltered` and `frontpageSoftFilter` are both empty,
then `defaultScoreModifiers` will be ignored

**Comment by soof-golan** [comment]
# Actually Testing The Thing

After a bit of manual testing against the dev DB this feature seems to work as intended 🎉

# No Multipliers

image

# 1000x to Biosecurity

image

**Comment by darkruby501** [comment]
Seems like a fine idea but I think is probably the wrong implementation. Introduces unnecessary complexity when we've got a bunch of functionality that's close.

You can get much this functionality via the Frontpage Tag Filters. If you want to go the "half the karma route", I'd be modifying that code, though my own preference would be to just make "-10" or something the default tag modifier for people. (Set that as what's created for new users and run a migration on the rest?)
